python/pyspark/pandas/series.py

```diff
@@ -2257,27 +2257,44 @@ def _interpolate(
             return self._psdf.copy()._psser_for(self._column_label)
 
         scol = self.spark.column
-        sql_utils = SparkContext._active_spark_context._jvm.PythonSQLUtils
-        last_non_null = PySparkColumn(sql_utils.lastNonNull(scol._jc))
-        null_index = PySparkColumn(sql_utils.nullIndex(scol._jc))
+        if is_remote():
+            from pyspark.sql.connect.functions import _invoke_function_over_columns
```

Comment:
will it minimize the changes if we put those functions into `python/pyspark/pandas/spark/functions.py`?

Reply:
Sounds good. Let me address the placement.

Reply:
Thanks, @zhengruifeng ! Just adjusted the comments.